File: polyglot/lua/cos_gate.lua
-- SPDX-License-Identifier: AGPL-3.0-or-later
-- lua lua/cos_gate.lua
local root = os.getenv("CREATION_OS_ROOT") or "."
local cmd = string.format('cd "%s" && exec make merge-gate', root:gsub('"', '\\"'))
local r = os.execute(cmd)
if r == true or r == 0 then os.exit(0) end
os.exit(1)

File: polyglot/elixir/cos_gate.exs
# SPDX-License-Identifier: AGPL-3.0-or-later
# mix run not required — elixir cos_gate.exs
root = System.get_env("CREATION_OS_ROOT") || "."
{_, code} = System.cmd("make", ["merge-gate"], cd: root, stderr_to_stdout: true)
System.halt(code)
